packages/wallet-service/src/api/healthcheck.ts

```diff
@@ -0,0 +1,145 @@
+    HealthcheckStatus,
+} from '@hathor/healthcheck-lib';
+import { getLatestHeight } from '@src/db';
+import fullnode from '@src/fullnode';
+import { closeDbConnection, getDbConnection } from '@src/utils';
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import { getRedisClient, ping } from '@src/redis';
+
+const mysql = getDbConnection();
+
+const HEALTHCHECK_MAXIMUM_HEIGHT_DIFFERENCE = Number(process.env.HEALTHCHECK_MAXIMUM_HEIGHT_DIFFERENCE ?? 5)
+
+const checkDatabaseHeight: HealthcheckCallbackResponse = async () => {
+  try {
+    const [currentHeight, fullnodeStatus] = await Promise.all([
+      getLatestHeight(mysql),
+      fullnode.getStatus()
+    ]);
+
+    const currentFullnodeHeight = fullnodeStatus['dag']['best_block']['height'];
+
+    if (currentFullnodeHeight - currentHeight < HEALTHCHECK_MAXIMUM_HEIGHT_DIFFERENCE) {
+      return new HealthcheckCallbackResponse({
+        status: HealthcheckStatus.PASS,
+        output: `Database and fullnode heights are within ${HEALTHCHECK_MAXIMUM_HEIGHT_DIFFERENCE} blocks difference`,
+      });
+    } else {
+      return new HealthcheckCallbackResponse({
+        status: HealthcheckStatus.FAIL,
+        output: `Database height is ${currentHeight} but fullnode height is ${currentFullnodeHeight}`,
+      });
+    }
+  } catch (e) {
+    console.error(e);
+
+    return new HealthcheckCallbackResponse({
+      status: HealthcheckStatus.FAIL,
+      output: `Error checking database and fullnode height: ${e.message}`,
+    });
+  }
+};
+
+const checkRedisConnection: HealthcheckCallbackResponse = async () => {
+  const client = getRedisClient();
+  try {
+    const pingResult = await ping(client);
+
+    if (pingResult === 'PONG') {
+      return new HealthcheckCallbackResponse({
+        status: HealthcheckStatus.PASS,
+        output: `Redis connection is up`,
+      });
+    } else {
+      return new HealthcheckCallbackResponse({
+        status: HealthcheckStatus.FAIL,
+        output: `Redis responded ping with invalid response: ${pingResult}`,
+      });
+    }
+  } catch (e) {
+    console.error(e);
+
+    return new HealthcheckCallbackResponse({
+      status: HealthcheckStatus.FAIL,
+      output: `Error checking redis connection: ${e.message}`,
+    });
+  }
+};
+
+const checkFullnodeHealth: HealthcheckCallbackResponse = async () => {
+  try {
+    const health = await fullnode.getHealth();
+
+    if (health['status'] === HealthcheckStatus.PASS) {
+      return new HealthcheckCallbackResponse({
+        status: HealthcheckStatus.PASS,
+        output: `Fullnode is healthy`,
+      });
+    } else if (health['status'] === HealthcheckStatus.WARN) {
+      return new HealthcheckCallbackResponse({
+        status: HealthcheckStatus.WARN,
+        output: `Fullnode has health warnings: ${health}`,
+      });
+    } else {
+      return new HealthcheckCallbackResponse({
+        status: HealthcheckStatus.FAIL,
+        output: `Fullnode is unhealthy: ${JSON.stringify(health)}`,
```

Comment:
We should review whether we should fail our healthcheck when the fullnode has just not received blocks for a while (which is a possible situation)